js/plugins/compat-oai/src/model.ts

```diff
@@ -98,6 +98,52 @@ export function toOpenAITool(tool: ToolDefinition): ChatCompletionTool {
   };
 }
 
+/**
+ * Checks if a content type is an image type.
+ * @param contentType The content type to check.
+ * @returns True if the content type is an image type.
+ */
+function isImageContentType(contentType?: string): boolean {
+  if (!contentType) return false;
+  return contentType.startsWith('image/');
+}
+
+/**
+ * Extracts the base64 data and content type from a data URL.
+ * @param url The data URL to parse.
+ * @returns The base64 data and content type, or null if invalid.
+ */
+function extractDataFromBase64Url(url: string): {
+  data: string;
+  contentType: string;
+} | null {
+  const match = url.match(/^data:([^;]+);base64,(.+)$/);
+  return (
+    match && {
+      contentType: match[1],
+      data: match[2],
+    }
+  );
+}
+
+/**
+ * Generates a filename from a content type.
+ * @param contentType The content type.
+ * @returns A filename with appropriate extension.
+ */
+function generateFilenameFromContentType(contentType: string): string {
+  const extensions: Record<string, string> = {
+    'application/pdf': 'pdf',
+    'application/msword': 'doc',
+    'application/vnd.openxmlformats-officedocument.wordprocessingml.document':
+      'docx',
+    'text/plain': 'txt',
+    'text/csv': 'csv',
+  };
+  const ext = extensions[contentType] || 'file';
```

Comment:
Instead of defaulting to `.file`, default to empty?

app/pdf -> file.pdf
text/csv -> file.csv
app/unknown -> file